includes/classes/Feature/BooleanSearchOperators.php

```diff
@@ -0,0 +1,339 @@
+ *
+ * @package ElasticPressLabs\Feature\BooleanSearchOperators
+ */
+class BooleanSearchOperators extends Feature {
+
+	/**
+	 * Initialize feature setting it's config
+	 */
+	public function __construct() {
+		$this->slug = 'boolean_search_operators';
+
+		$this->title = esc_html__( 'Boolean Search Operators', 'elasticpress-labs' );
+
+		$this->requires_install_reindex = false;
+		$this->default_settings         = [
+			'active' => false,
+		];
+
+		parent::__construct();
+	}
+
+	/**
+	 * Setup Feature Functionality
+	 */
+	public function setup() {
+		$settings = $this->get_settings();
+
+		if ( $settings['active'] ) {
+			/** Features Class @var Features $features */
+			$features = Features::factory();
+
+			/** Search Feature @var Feature\Search\Search $search */
+			$search = $features->get_registered_feature( 'search' );
+
+			if ( ! $search->is_active() && $this->is_active() ) {
+				$features->deactivate_feature( $this->slug );
+
+				return false;
+			}
+
+			add_filter( 'ep_elasticpress_enabled', [ $this, 'integrate_boolean_search_operators' ], 10, 2 );
+		}
+	}
+
+	/**
+	 * Hook up the boolean operators query if it is enabled
+	 *
+	 * @hook   ep_elasticpress_enabled
+	 *
+	 * @param  {bool}     $enabled Whether to integrate with Elasticsearch or not
+	 * @param  {WP_Query} $query WP_Query to evaluate
+	 *
+	 * @return bool
+	 */
+	public function integrate_boolean_search_operators( $enabled, $query ) {
+		if ( ! $enabled ) {
+			return false;
+		}
+
+		if ( true === $query->query_vars['ep_integrate'] &&
+			( $this->is_active() || true === $query->query_vars['ep_boolean_operators'] ) ) {
+
+			\add_filter( 'ep_formatted_args_query', [ $this, 'replace_query_if_boolean' ], 999, 4 );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if a search query uses boolean operators and switch queries accordingly
+	 *
+	 * @hook  ep_formatted_args_query
+	 *
+	 * @param {array}  $query         Current query
+	 * @param {array}  $query_vars    Query variables
+	 * @param {string} $search_text   Search text
+	 * @param {array}  $search_fields Search fields
+	 *
+	 * @return array
+	 */
+	public function replace_query_if_boolean( $query, $query_vars, $search_text, $search_fields ) {
+
+		if ( $this->query_uses_boolean_operators( $search_text ) ) {
+
+			$simple_query = array(
+				'simple_query_string' => array(
+
+					/**
+					 * Filter the fields to use in boolean operator searches
+					 *
+					 * @hook    ep_labs_boolean_operators_fields
+					 *
+					 * @param   {array} $search_fields
+					 * @param   {array}  $query_vars    Query variables
+					 * @param   {string} $search_text   Search text modified to replace tokens
+					 * @param   {array}  $search_fields Search fields
+					 * @param   {array}  $query         The original query
+					 *
+					 * @return  {array} New fields
+					 */
+					'fields'           => \apply_filters( 'ep_labs_boolean_operators_fields', $search_fields, $query_vars, $search_text, $query ),
```

Comment:
can we add `@since` tags to these new filters, @Rahmon?

Reply:
added